src/graphqlProtocol/resolver.ts

```diff
@@ -562,7 +562,7 @@ export const handleSubscriptionPermission = async (
     where: permissionWhere,
     lookupPermissionEntity,
   } = await buildActionPermissionFilter(
-    subPermissions,
+    () => subPermissions,
```

Comment:
`subPermissions` could be a permission object or a function returning a permission.
The additional function wrapper will break the tests if a permission function is provided.

Reply:
in this case subpermission is already an array because of the concatenation that's done a few lines above
```ts
  const subPermissions = ([] as Permission[]).concat(
    ...Object.values(permissionsMap.subscriptions as Permission | Permission[]),
  );
```

Reply:
so wrapping it in a function shouldn't be a problem here